srt-protocol/src/listener/session.rs

```diff
@@ -0,0 +1,89 @@
+use std::{net::SocketAddr, time::Instant};
+
+use crate::{
+    packet::Packet, protocol::pending_connection::listen::Listen, settings::ConnInitSettings,
+};
+
+use super::*;
+
+#[derive(Debug)]
+pub enum SessionState {
+    Pending(Listen),
+    Rejecting(Option<(Packet, SocketAddr)>),
+    Dropping,
+    Open,
+}
+
+impl SessionState {
+    pub fn new_oending(settings: ConnInitSettings) -> Self {
```

Comment:
Is this supposed to be `new_pending`?

Reply:
> @robertream, here's an initial review. I haven't reviewd all the files yet, but I had limited time and figured I'd at least send back my thoughts.
> 
> I love AccessResponse being an Input. Much cleaner than what I had.
> 
> Can we delete Multiplex yet?

Deleting Multiplex is the end goal, at least shortly after this PR if not part of it. I haven't looked into if the effort is non-trivial. I'm less concerned about getting this PR perfect, we can clean it up as we iterate, we don't have many people using the library... yet. :)

> There's some issue with large allocations in some of your newest tests
> 
> Rust users are used to the builder pattern, no reason to hide it from them by returning a builder form SrtSocket's new--can we just call that function `builder` to make it clear what's going on? I've seen that pattern in, for example, Quinn and I think it's slick. And We should call the builder `SrtSocketBuilder`, as it is a builder.

I am fine with this, I have a bias against using programming design pattern names in types. Anything in my domain model that ends with "or", "er" or "ar", I am highly skeptical of, unless it's essential to the domain language (e.g. the SRT spec actually mentions Sender/Receiver Listener/Caller). Since Builder is a convention in Rust, I'll concede this point: "when in Rome...". I'm glad you approve of factory methods for builders as a member of the struct that is being built. I'm fine with naming the method "builder".

Reply:
yup